app/graphql/types/project_id_filter_type.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module Types
+  class ProjectIdFilterType < Types::BaseInputObject
+    argument :all_of, [GraphQL::Types::ID], required: false
+    argument :any_of, [GraphQL::Types::ID], required: false
+
+    validates required: { one_of: %i[all_of any_of], message: 'May use either allOf  or anyOf, but not both.' }
```

Comment:
Grammar nazi here - There's two spaces in the message after `allOf` :no_mouth: 

That aside, is the custom message required? We could use the default one if it has one.

Reply:
Default message is **ProjectIdFilter has the wrong arguments**. I think it would be more clear to use the exemplified custom message. Surely, after removing blank spaces. :slightly_smiling_face: 
Same for `technology_id_filter_type`.